videorender.cpp

```diff
@@ -155,65 +177,114 @@ void cVideoRender::DestroyFrameBuffers(void)
 		m_destroyCurrentlyDisplayed = true;
 }
 
+/**
+ * Destroy all frame buffers, except the currently displayed one.
+ */
+void cVideoRender::DestroyPipFrameBuffers(void)
+{
+	for (int i = 0; i < RENDERBUFFERS; ++i) {
+		if (!m_pipBuffer[i].IsDirty())
+			continue;
+
+		m_pipBuffer[i].Destroy();
+	}
+
+	m_numPipBuffers = 0;
+	m_enqueuePipBufferIdx = 0;
+	m_pCurrentlyPipDisplayed = nullptr;
+}
+
+struct sRect {
+	uint64_t x;
+	uint64_t y;
+	uint64_t w;
+	uint64_t h;
+};
+
+/**
+ * Fits the video frame into a given area
+ *
+ * @param frame        AVFrame with frame dimensions and aspect ratio information
+ * @param dispX        x offset of video area
+ * @param dispY        y offset of video area
+ * @param dispWidth    width of video area
+ * @param dispHeight   height of video area
+ *
+ * @returns            the new computed video area or the given area if no frame was given
+ */
+static sRect ComputeFittedRect(AVFrame *frame, uint64_t dispX, uint64_t dispY, uint64_t dispWidth, uint64_t dispHeight)
+{
+	if (!frame || dispWidth == 0 || dispHeight == 0)
+		return { dispX, dispY, dispWidth, dispHeight };
+
+	double frameWidth = frame->width > 0 ? static_cast<double>(frame->width) : 1.0;
```

Comment:
Is there a reason using static_cast?
```suggestion
	double frameWidth = frame->width > 0 ? frame->width : 1;
```